packages/api/src/auth.js

```diff
@@ -168,6 +169,17 @@ async function tryWeb3ApiToken (token, env) {
     // we have a web3 api token, but it's no longer valid
     throw new TokenNotFoundError()
   }
+
+  if (apiToken.isDeleted) {
+    const isBlocked = await checkIsTokenBlocked(apiToken, env)
+
+    if (isBlocked) {
+      throw new TokenBlockedError()
+    } else {
+      throw new TokenNotFoundError()
+    }
+  }
+
```

Comment:
The `checkIsTokenBlocked` check is good. Let's simplify this PR to keep it so that the db client `getKey` method doen't return deleted keys, and just add the blocked check once we determine that the key the user has sent us is not valid.

```suggestion
    // we have a web3 api token, but it's no longer valid
    const isBlocked = await checkIsTokenBlocked(apiToken, env)
    if (isBlocked) {
      throw new TokenBlockedError()
    } else {
      throw new TokenNotFoundError()
    }
  }

```

Reply:
@olizilla is there any performance benefit to filtering by isDeleted?  the lookup to fetch based on the token irrespective of whether or not it has been deleted i'm assuming is negligible.  the downside to your approach is that we have to read from the auth_key table twice in all scenarios.

Reply:
Ah, the bit I missed was we need the `auth_key` here to check if it is blocked. My propsal doesn't work. My only feels here is that It would be nice to not have to bring back all deleted keys here on the happy path, and the scheme feels a like it needs a refactor, but that is beyond the scope of this PR. I see the equivalent PR already got merged in nft.storage, so for the sake of consistency this should go in as is.

Reply:
@olizilla i may be wrong, but i believe the query in `dbClient.getKey` _only returns 1 key_ by virtue of `.eq('keys.secret', secret)`.